projects/Mallard/src/helpers/files.ts

```diff
@@ -30,41 +31,68 @@ export const fileIsIssue = (file: File): file is IssueFile =>
     file.type === 'issue'
 
 export const ensureDirExists = (dir: string): Promise<void> =>
-    RNFetchBlob.fs.mkdir(dir).catch(() => Promise.resolve())
+    RNFS.mkdir(dir).catch(() => Promise.resolve())
 
 /*
 We always try to prep the file system before accessing issuesDir
 */
-export const prepFileSystem = (): Promise<void> =>
-    ensureDirExists(FSPaths.issuesDir).then(() =>
-        ensureDirExists(`${FSPaths.issuesDir}/daily-edition`),
-    )
+export const prepFileSystem = async (): Promise<void> => {
+    await ensureDirExists(FSPaths.issuesDir)
+    await ensureDirExists(`${FSPaths.issuesDir}/daily-edition`)
+    await ensureDirExists(FSPaths.downloadRoot)
+}
 
 export const getJson = <T extends any>(path: string): Promise<T> =>
-    RNFetchBlob.fs.readFile(path, 'utf8').then(d => JSON.parse(d))
+    RNFS.readFile(path, 'utf8').then(d => JSON.parse(d))
 
-export const downloadNamedIssueArchive = async (
-    localIssueId: Issue['localId'],
-    assetPath: string,
-) => {
+export const downloadNamedIssueArchive = async ({
+    localIssueId,
+    assetPath,
+    filename,
+    withProgress,
+}: {
+    localIssueId: Issue['localId']
+    assetPath: string
+    filename: string
+    withProgress: boolean
+}) => {
     const apiUrl = await getSetting('apiUrl')
     const zipUrl = `${apiUrl}${assetPath}`
-    const returnable = RNFetchBlob.config({
-        fileCache: true,
-        overwrite: true,
-        IOSBackgroundTask: true,
-    }).fetch('GET', zipUrl)
-    return {
-        promise: returnable.then(async res => {
-            // Ensure issue is removed from the cache on completion
-            const { clear } = withCache('issue')
-            clear(localIssueId)
-            await prepFileSystem()
-            await ensureDirExists(FSPaths.issueRoot(localIssueId))
-            return res
-        }),
-        cancel: returnable.cancel,
-        progress: returnable.progress,
+    const downloadFolderLocation = FSPaths.downloadIssueLocation(localIssueId)
+    await prepFileSystem()
+    await ensureDirExists(FSPaths.issueRoot(localIssueId))
+    await ensureDirExists(downloadFolderLocation)
+
+    try {
+        const returnable = RNFS.downloadFile({
+            fromUrl: zipUrl,
+            toFile: `${downloadFolderLocation}/${filename}`,
+            background: true,
+            begin: () => console.log('start download'),
```

Comment:
Required to make progress work..... 🤦

Reply:
Strange! does passing empty function body works?